File: src/test/java/com/baulsupp/oksocial/i9n/SurveyMonkeyTest.java
package com.baulsupp.oksocial.i9n;

import com.baulsupp.oksocial.Main;
import com.baulsupp.oksocial.services.surveymonkey.SurveyMonkeyAuthInterceptor;
import com.baulsupp.oksocial.services.surveymonkey.SurveyMonkeyToken;
import com.google.common.collect.Lists;
import org.junit.jupiter.api.Test;

import static com.google.common.collect.Lists.newArrayList;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class SurveyMonkeyTest {

  private Main main = new Main();
  private TestOutputHandler output = new TestOutputHandler();
  private TestCompletionVariableCache completionCache = new TestCompletionVariableCache();
  private TestCredentialsStore credentialsStore = new TestCredentialsStore();

  {
    main.outputHandler = output;
    main.completionVariableCache = completionCache;
    main.credentialsStore = credentialsStore;
  }

  @Test public void completeEndpointWithReplacements() throws Throwable {
    main.arguments = newArrayList("https://api.surveymonkey.net/");
    main.urlComplete = true;
    completionCache.store("surveymonkey", "surveys", Lists.newArrayList("AA", "BB"));
    credentialsStore.storeCredentials(new SurveyMonkeyToken("", ""),
        new SurveyMonkeyAuthInterceptor().serviceDefinition());

    main.run();

    assertEquals(Lists.newArrayList(), output.failures);
    assertEquals(1, output.stdout.size());
    assertTrue(output.stdout.get(0).contains("/v3/surveys/AA/details"));
    assertTrue(output.stdout.get(0).contains("/v3/surveys/BB/details"));
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/i9n/SurveyMonkeyTest.kt
package com.baulsupp.oksocial.i9n

import com.baulsupp.oksocial.DefaultToken
import com.baulsupp.oksocial.Main
import com.baulsupp.oksocial.authenticator.oauth2.Oauth2Token
import com.baulsupp.oksocial.output.TestOutputHandler
import com.baulsupp.oksocial.services.surveymonkey.SurveyMonkeyAuthInterceptor
import okhttp3.Response
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class SurveyMonkeyTest {

  private val main = Main()
  private val output = TestOutputHandler<Response>()
  private val completionCache = TestCompletionVariableCache()
  private val credentialsStore = TestCredentialsStore()

  init {
    main.outputHandler = output
    main.completionVariableCache = completionCache
    main.credentialsStore = credentialsStore
  }

  @Test
  fun completeEndpointWithReplacements() {
    main.arguments = mutableListOf("https://api.surveymonkey.net/")
    main.urlComplete = true
    completionCache["surveymonkey", "survey"] = listOf("AA", "BB")
    credentialsStore.set(SurveyMonkeyAuthInterceptor().serviceDefinition(), DefaultToken.name, Oauth2Token(""))

    main.run()

    assertEquals(mutableListOf(), output.failures)
    assertEquals(1, output.stdout.size)
    assertTrue(output.stdout[0].contains("/v3/surveys/AA/details"))
    assertTrue(output.stdout[0].contains("/v3/surveys/BB/details"))
  }
}
